src/wagtail_ai/models.py

```diff
@@ -0,0 +1,106 @@
+import uuid
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from wagtail.search import index
+
+DEFAULT_PROMPTS = [
+    {
+        "uuid": "fe029b02-833e-49d6-8002-14619962946a",  # This is a static UUID used to identify the prompt
+        "label": "AI Correction",
+        "description": "Correct grammar and spelling",
+        "prompt": (
+            "You are assisting a user in writing content for their website. "
+            "The user has provided some text (following the colon). "
+            "Return the provided text but with corrected grammar, spelling and punctuation. "
+            "Do not add additional punctuation, quotation marks or change any words:"
+        ),
+        "method": "replace",
+    },
+    {
+        "uuid": "cc4805e3-abb6-4a09-b71c-b5543af34eb1",  # This is a static UUID used to identify the prompt
+        "label": "AI Completion",
+        "description": "Get help writing more content based on what you've written",
+        "prompt": (
+            "You are assisting a user in writing content for their website. "
+            "The user has provided some initial text (following the colon). "
+            "Assist the user in writing the remaining content:"
+        ),
+        "method": "append",
+    },
+]
+
+
+class Prompt(models.Model, index.Indexed):
+    class Method(models.TextChoices):
+        REPLACE = "replace", _("Replace content")
+        APPEND = "append", _("Append after existing content")
+
+    uuid = models.UUIDField(default=uuid.uuid4, editable=False)
+    label = models.CharField(max_length=50)
+    description = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text=_(
+            "The prompt description, displayed alongside its label on the settings listing."
+        ),
+    )
+    prompt = models.TextField(  # noqa: DJ001
+        null=True,
+        blank=False,
+        help_text=_(
+            "The prompt text sent to the Large Language Model (e.g. ChatGPT) to generate content."
+        ),
+    )
+    method = models.CharField(
+        max_length=25,
+        choices=Method.choices,
+        help_text=_("The method used for processing the responses to the prompt."),
+    )
+
+    search_fields = [
+        index.SearchField("label"),
+        index.SearchField("description"),
+        index.SearchField("prompt"),
+    ]
+
+    def __str__(self):
+        return self.label
+
+    def is_default_prompt(self) -> bool:
```

Comment:
**Nitpick:** Could be a property, it does not access the database or does any other expensive operations.

Also, it's name does read like a property `prompt.is_default`.

That said, it could probably just be called `is_default` as otherwise you get a repetition, i.e. `prompt.is_default_prompt`.